rollup/internal/controller/relayer/l2_relayer.go

```diff
@@ -327,6 +327,12 @@ func (r *Layer2Relayer) ProcessPendingBatches() {
 				log.Error("Failed to get parent batch header", "index", batch.Index-1, "error", err)
 				return
 			}
+
+			if types.RollupStatus(parentBatch.RollupStatus) == types.RollupCommitFailed {
```

Comment:
what if the parent batch is going to failed but not failed yet? The rollup status will be submitted.

Reply:
We will have multiple failing commit txs, the first tx fails with some reason, and the others fail with `incorrect parent batch hash`. But at some point, this line short-circuits the process, and we won't submit any more failing commit txs.

Also, during normal operation, there will only be 1 new batch / 2-4 minutes, so this mechanism will work well there.

Reply:
make sense, but I think it;s acceptable.
 
 we will still submit this batch,
 we only get 5 batches each time
 next time it will probably already become failed.